### PR TITLE
Shutdown server on SIGTERM

### PR DESCRIPTION
**Pull request by BadrTad**
Now, server terminates on` SIGTERM (Ctrl+C)` instead of `RETURN`.

### CODE REVIEW

**Review by tuomas-p — DISMISSED** [review]
Looks clean to me

File: be2-scala/src/main/scala/ch/epfl/pop/Server.scala [hunk 1]
```diff
@@ -67,20 +68,24 @@ object Server {
...
-          logger.info("Server terminated !")
-          system.terminate()
-          typedSystem.terminate()
-        }) // and shutdown when done
-
-      Behaviors.empty
+      /* Shutting down */
+      val shutdownListener = new Thread() {
+        override def run(): Unit = {
+          logger.warning("shutdown in 5s ");
```
Comment (pierluca) [inline be2-scala/src/main/scala/ch/epfl/pop/Server.scala:74]: is this log still warranted?

File: be2-scala/src/main/scala/ch/epfl/pop/Server.scala [hunk 2]
```diff
@@ -67,20 +68,24 @@ object Server {
...
+                system.terminate()
+                typedSystem.terminate()
+              }) // and shutdown when done
+          } catch {
+                case  e: InterruptedException =>   logger.warning("Server shutting thread was interrupted !")
+          }
+        }
+      };
+    Runtime.getRuntime().addShutdownHook(shutdownListener);
+    Behaviors.empty
```
Comment (pierluca) [inline be2-scala/src/main/scala/ch/epfl/pop/Server.scala:87]: is the indent correct here? 🤔

File: be2-scala/src/main/scala/ch/epfl/pop/Server.scala (hunk 1 above)
Comment (BadrTad) [reply be2-scala/src/main/scala/ch/epfl/pop/Server.scala:74]: Nope

File: be2-scala/src/main/scala/ch/epfl/pop/Server.scala (hunk 2 above)
Comment (BadrTad) [reply be2-scala/src/main/scala/ch/epfl/pop/Server.scala:87]: Does not seem like, however I am sure my IDE is configured with `.editorconfig  `

**Review by BadrTad — COMMENTED** [review]
The changes from your comments have been applied ! 
Thanks for review

**Review by tuomas-p — DISMISSED** [review]
Seems all good as this is for merge conflicts?

**Review by BadrTad — COMMENTED** [review]
All the comments have been addressed. I think this PR is ready to be merged

**Review by BadrTad — COMMENTED** [review]
This fixes the git history and apply a sbt assembly custom  merge to resolve internal metals duplicates

File: be2-scala/build.sbt [hunk 3]
```diff
@@ -1,84 +1,89 @@
...
+  "sonar.scala.coverage.reportPaths" -> "./target/scala-2.13/scoverage-report/scoverage.xml",
+  "sonar.scala.scapegoat.reportPaths" -> "./target/scala-2.13/scapegoat-report/scapegoat.xml"
+)
+
+assemblyMergeStrategy in assembly := {
+    case PathList("module-info.class") => MergeStrategy.discard
+    case PathList("reference.conf") => MergeStrategy.concat
+    case PathList("META-INF","MANIFEST.MF") => MergeStrategy.discard
+    case _ => MergeStrategy.defaultMergeStrategy("")
+}
```
[the remark this answers is not in the document]
Comment (pierluca) [reply be2-scala/build.sbt:46]: how come the changes seem to affect the whole file? 
Were the line endings changed? (LF vs. CRLF)

**Review by pierluca — APPROVED** [review]
Minor remark, but looks good to me !

**Pull request merged**